Update Worldspace Defs + some General Cleanup

robertgk2017 opened the pull request:
Decoded WRLD\CLSZ for FO4/FO76/SF
Decoded WRLD\VISI for FO76

Moved common large array definitions out of the game files and into DefinitionsCommon
Cleaned up LargeRefs/MHDT/OFST/CLSZ/VISI across the various games
Added wbNeverShow and wbSimpleRecords splits for all of these large array sub-records for performance.

Removed a useless wbSImpleRecords split of the entirety of WRLD in TES4/FO3/FNV as both sides we're completely identical.

Big thanks to Eck, Evan, and John for feedback on this one :)

robertgk2017 committed:
Update Worldspace Arrays

Decoded/Defined CLSZ, and VISI.  As well as cleaned up the definitions for Large Refs, MHDT, OFST, CLSZ, VISI across the games and moved definitions to the Common File with wbSimpleRecords conditions.

robertgk2017 committed:
Change wbRNAM to wbLargeReferences

robertgk2017 committed:
Update TES4, FO4, FNV

the entire WRLD record was duplicated in a wbSimpleRecords if/else which was entirely identical except for OFST, which already natively handles wbSimpleRecords.

Also added the same df Flags the other games have/

robertgk2017 committed:
Sorted Large Refs

Tested loading WRLDs with lots of overrides and lots of Large Refs.  namely Commonwealther and Tamriel in FO4/SE.  found no discernable difference in record load times.

robertgk2017 committed:
Update

Added dfFastAssigned to the nested wbArrays under the top Arrays, seems to further speed up WRLD viewing, maybe?

Also implemented a wbRemoveCLSZ for FO4/FO76

Adjusted the wbWRLDAfterLoad OBND fixer function to be named wbFixWorldOBND to allow nesting under an actual wbWRLDAfterLoad that calls it and the wbRemoveOFST/CLSZ/VISI

robertgk2017 committed:
Update

accidentally dropped this OBND change in the wrong branch

robertgk2017 committed:
Update wbDefinitionsCommon.pas

Forgot a cpIgnore on wbLargeReferences

robertgk2017 committed:
Reimplement a remove Large ref function

This time conditioned on -DontRemoveOffsetData like the other giant Worldspace arrays., and similarly limited to the primary game master, for Skyrim, Fallout 4, and Fallout 76.

robertgk2017 committed:
Update wbDefinitionsCommon.pas

Add Summary Key to wbLargeReferences

robertgk2017 committed:
Update wbDefinitionsCommon.pas

And cleanup the summary key

robertgk2017 committed:
Merge branch 'dev-4.1.6' of https://github.com/TES5Edit/TES5Edit into Main

robertgk2017 committed:
Resolve Merge Conflict

And added the existing WRLD MHDT to SF1.  still need to decode the different CELL MHDT.

robertgk2017 committed:
Update wbDefinitionsSF1.pas

Add CELL MHDT

robertgk2017 committed:
Move Cell MHDT to Common

Figured out you can do an Else Begin in an If Else.  Move SF CELL MHDT to common so it's all together under one definition.

robertgk2017 committed:
Import TES3 Sigs to DefinitionSigs

robertgk2017 committed:
Import TES4 Sigs to DefinitionSigs

robertgk2017 committed:
Update wbDefinitionsTES5.pas

Merge the split Define... procedures into one DefineTES5.  Potentially sloppy had to move some things around till it worked.  Will be an ongoing task to clean up/re-organize the file.

robertgk2017 committed:
Update wbDefinitionsTES5.pas

Refactor for TES5 progress

Moved WRLD to the bottom to begin sorting the wbRecords by record name and moved the wbWRLDAfterLoad and subordinate procedures that are WRLD exclusive to be placed right next to the wbRecord.

robertgk2017 committed:
Update wbDefinitionsTES5.pas

Adjusted indentation on WRLD, moved WOOP and WTHR to next to WRLD.  next is to figure out how to move the Weather functions next to WTHR.

robertgk2017 committed:
Reconstruct PR

robertgk2017 committed:
Organize DefsCommon variables by type

Also Move several giant wbFlags defs to common instead of being copypasta'd across the games.

robertgk2017 committed:
Merge branch 'dev-4.1.6' of https://github.com/TES5Edit/TES5Edit into Main

robertgk2017 committed:
Update wbDefinitionsTES4.pas

Move WRLD to be the last wbRecord in the DefineTES4 procedure, and tidy it up a bit more.

robertgk2017 committed:
Update wbDefinitionsCommon.pas

Remove the simple records splitter for large refs, can't be breaking mapped formids.

robertgk2017 committed:
Implement an actually working Remove Large Refs, this time actually confirmed working.

robertgk2017 committed:
Merge branch 'dev-4.1.6' of https://github.com/TES5Edit/TES5Edit into Main

robertgk2017 committed:
Update wbDefinitionsCommon.pas

robertgk2017 committed:
Update wbDefinitionsCommon.pas

dsp2oo3 commented on Core/wbDefinitionsCommon.pas:
```diff
@@ -437,6 +518,76 @@ function wbNormalizeToRange(aMin, aMax: Extended): TwbFloatNormalizer;
+    {0x00000004} { 2} 'Unknown 2',
+    {0x00000008} { 3} 'Unknown 3',
+    {0x00000010} { 4} 'Unknown 4',
+    {0x00000020} { 4} 'Unknown 5',
```
I understand it's commented out, but isn't { 4} supposed to be { 5} for 'Unknown 5'?

robertgk2017 replied:
heh that's been like that for who knows how long.  I just copypasta'd it over to Common.

robertgk2017 committed:
Update wbDefinitionsCommon.pas

robertgk2017 committed:
Update wbDefinitionsTES5.pas

NPC_ \ WNAM is not "Worn Armor" it's Skin in the CK.

dsp2oo3 commented on Core/wbDefinitionsTES5.pas:
```diff
@@ -10760,7 +10760,7 @@ procedure DefineTES5m;
     wbSPLOs,
     wbDEST,
-    wbFormIDCk(WNAM, 'Worn Armor', [ARMO], False, cpNormal, False),
+    wbFormIDCk(WNAM, 'Skin', [ARMO], False, cpNormal, False),
```
Technically, it defines the default armor (because body mesh is an armor) occupying slot 32 when character either wears no clothes or clothing pieces that do not have their own body meshes. Suggestion: "Skin (Default Body Armor)"

robertgk2017 replied:
It's labeled Skin in the CK.  and you only use this on non-humanoid creatures with the Skin ARMO that is the actual creature .nif, for example Draugr.   That's also not the default "armor".  Thats the skin on the RACE record,  Skin on an NPC_ is an override of the RACE Skin.

robertgk2017 committed:
Update wbDefinitionsCommon.pas

Remove 3 unused variables from the WRLD Array Column Functions.

robertgk2017 committed:
Update wbDefinitionsTES5.pas

ARMO: Add Visible When Distant flag
ACHR: Add Visible When Distant Flag
ACHR: Add XLOD sub-record
NAVM: Cover Height Comment was backwards.

robertgk2017 committed:
Merge branch 'dev-4.1.6' of https://github.com/TES5Edit/TES5Edit into Main
